utils/s3/client.cc

```diff
@@ -169,7 +169,16 @@ void client::authorize(http::request& req) {
 
 future<> client::make_request(http::request req, http::experimental::client::reply_handler handle, http::reply::status_type expected) {
     authorize(req);
-    return _http.make_request(std::move(req), std::move(handle), expected);
+    auto sg = current_scheduling_group();
+    auto it = _https.find(sg);
+    if (it == _https.end()) [[unlikely]] {
+        auto factory = std::make_unique<dns_connection_factory>(_host, _cfg->port, _cfg->use_https);
+        it = _https.emplace(std::piecewise_construct,
+            std::forward_as_tuple(sg),
+            std::forward_as_tuple(std::move(factory))
+        ).first;
+    }
+    return it->second.make_request(std::move(req), std::move(handle), expected);
 }
```

Comment:
would be nice to test this. a regular write workload, and observe flush and compaction rate.